pkg/solidity-utils/contracts/helpers/BufferHelpers.sol

```diff
@@ -0,0 +1,67 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+pragma solidity ^0.8.24;
+
+import { IERC4626 } from "@openzeppelin/contracts/interfaces/IERC4626.sol";
+
+import { PackedTokenBalance } from "./PackedTokenBalance.sol";
+
+library BufferHelpers {
+    using PackedTokenBalance for bytes32;
+
+    /**
+     * @dev Underlying surplus is the amount of underlying that need to be wrapped for the buffer to be rebalanced.
+     * For instance, consider the following scenario:
+     * - buffer balances: 2 wrapped and 10 underlying
+     * - wrapped rate: 2
+     * - normalized buffer balances: 4 wrapped as underlying (2 wrapped * rate) and 10 underlying
+     * - surplus of underlying = (10 - 4) / 2 = 3 underlying
+     * We need to wrap 3 underlying tokens to consider the buffer rebalanced.
+     * - 3 underlying = 1.5 wrapped
+     * - final balances: 3.5 wrapped (2 existing + 1.5 new) and 7 underlying (10 existing - 3)
+     */
+    function getBufferUnderlyingSurplus(bytes32 bufferBalance, IERC4626 wrappedToken) internal view returns (uint256) {
+        uint256 underlyingBalance = bufferBalance.getBalanceRaw();
+
+        uint256 wrappedBalanceAsUnderlying = 0;
+        if (bufferBalance.getBalanceDerived() > 0) {
+            wrappedBalanceAsUnderlying = wrappedToken.convertToAssets(bufferBalance.getBalanceDerived());
+        }
+
+        uint256 surplus = 0;
+        if (underlyingBalance > wrappedBalanceAsUnderlying) {
+            unchecked {
+                surplus = (underlyingBalance - wrappedBalanceAsUnderlying) / 2;
```

Comment:
```
underlyingBalance = 2
wrappedBalanceAsUnderlying = 1
```
surplus may overflow in this case but maybe it isn't problem

Reply:
In that case surplus would be 0, which should be fine.